sharktank_models/llama3.1/test_llama.py

```diff
@@ -0,0 +1,111 @@
+import iree.compiler
+import iree.runtime
+import numpy
+import os
+import pytest
+import torch
+
+page_size = 12288
+block_size = 16
+
+
+@pytest.fixture
+def llama_mlir():
+    dirname = os.path.dirname(__file__)
+    return f"{dirname}/assets/toy_llama.mlir"
+
+
+@pytest.fixture
+def llama_irpa():
+    dirname = os.path.dirname(__file__)
+    return f"{dirname}/assets/toy_llama.irpa"
+
+
+@pytest.fixture
+def compile_llama_cpu(llama_mlir):
+    return iree.compiler.compile_file(
+        llama_mlir,
+        extra_args=[
+            "--iree-hal-target-device=llvm-cpu",
+            "--iree-llvmcpu-target-cpu=host",
+        ],
+    )
+
+
+@pytest.fixture
+def compile_llama_gfx1100(llama_mlir):
+    return iree.compiler.compile_file(
+        llama_mlir,
+        extra_args=["--iree-hal-target-device=hip", "--iree-hip-target=gfx1100"],
+    )
+
+
+class ToyLlama:
+    def __init__(self, compiled, device_id, irpa):
+        paramIndex = iree.runtime.ParameterIndex()
+        paramIndex.load(irpa)
+        provider = paramIndex.create_provider("model")
+
+        self.instance = iree.runtime.VmInstance()
+        self.device = iree.runtime.get_device(device_id)
+        self.config = iree.runtime.Config(device=self.device)
+        self.parameters = iree.runtime.create_io_parameters_module(
+            self.instance, provider
+        )
+        self.hal = iree.runtime.create_hal_module(self.instance, self.device)
+        self.binary = iree.runtime.VmModule.copy_buffer(self.instance, compiled)
+        self.modules = iree.runtime.load_vm_modules(
+            self.parameters, self.hal, self.binary, config=self.config
+        )
+
+        self.prefill = self.modules[-1].prefill_bs1
+        self.decode = self.modules[-1].decode_bs1
+
+
+@pytest.fixture
+def toy_llama_cpu(compile_llama_cpu, llama_irpa):
+    return ToyLlama(compile_llama_cpu, "local-sync", llama_irpa)
```

Comment:
May want to use `local-task` instead of `local-sync` for the runtime device, depending on how long the test takes to run on CPU and how much multithreading influences the test behavior.

Could also use keyword arguments instead of positional arguments given how arbitrary these variables/strings look:

```suggestion
@pytest.fixture
def toy_llama_cpu(compile_llama_cpu, llama_irpa):
    return ToyLlama(compiled=compile_llama_cpu, device_id="local-task", irpa=llama_irpa)
```

Reply:
Updated.